Python/bltinmodule.c

```diff
@@ -2357,7 +2357,7 @@ builtin_sum_impl(PyObject *module, PyObject *iterable, PyObject *start)
        Assumes all inputs are the same type.  If the assumption fails, default
        to the more general routine.
     */
-    if (PyLong_CheckExact(result)) {
+    if (PyLong_CheckExact(result) || PyBool_Check(result)) {
```

Comment:
While it's consistent, it's a little strange to try to optimize for the case of the start value being bool; summing an iterable of bools is normal, but explicitly passing False as the start value surely isn't, and it means slowing (if trivially) progression to other fast paths, and the slow path, to check an unlikely scenario.

It also means a slightly behavioral change: Previously, sum with a start of False/True and an empty iterable would return False/True unchanged.

I'm thinking maybe it's fine to end up on the slow path for a start of False/True (presumably a nearly non-existent use case, but one which if used, might depend on the current behavior), since it's so unlikely, and the real savings is seen when the iterable is of bools.

Reply:
Good point!